includes/wc-core-functions.php

```diff
@@ -1491,7 +1491,7 @@ function wc_get_shipping_method_count( $include_legacy = false ) {
 		return absint( $transient_value['value'] );
 	}
 
-	$method_count = absint( $wpdb->get_var( "SELECT COUNT(*) FROM {$wpdb->prefix}woocommerce_shipping_zone_methods" ) );
+	$method_count = absint( $wpdb->get_var( "SELECT COUNT(*) FROM {$wpdb->prefix}woocommerce_shipping_zone_methods WHERE is_enabled=1" ) );
```

Comment:
We should tie this behavior to an argument so that this isn't a breaking change.

Reply:
From my understanding of how this method currently works, when `$include_legacy` is set to true it counts:
* all shipping methods that support shipping zones (no matter if they are enabled or not) +
* only the legacy shipping methods that are enabled (it filters them [here](https://github.com/woocommerce/woocommerce/blob/master/includes/wc-core-functions.php#L1516)).

My concern is that even if we add an argument in order to control whether disabled shipping methods must be filtered out or not, it will be breaking the current behavior because results were inconsistent when `$include_legacy` was true.

Is that right? What would be the correct way to proceed here? I'm not really used to this code, so I might be misunderstanding it.

Reply:
I think that given the legacy nature of the `$include_legacy` param, perhaps we could keep that behavior the same and document that the second `$enabled_only` does not affect the legacy methods.

I'm just concerned about the backwards incompatibility of the change is all. I agree it should be fixed, but since it has been this way for so long, I'm not sure it should be. What do you think?

Reply:
Sounds good. I rebased this PR and updated it with the feedback. Let me know how it looks.